src/main/java/org/kiwiproject/dropwizard/poller/ClientPoller.java

```diff
@@ -0,0 +1,492 @@
+package org.kiwiproject.dropwizard.poller;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotNull;
+import static org.kiwiproject.base.KiwiStrings.format;
+import static org.kiwiproject.concurrent.Async.doAsync;
+import static org.kiwiproject.concurrent.Async.withMaxTimeout;
+import static org.kiwiproject.jaxrs.KiwiResponses.closeQuietly;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.setup.Environment;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
+import org.kiwiproject.dropwizard.poller.metrics.ClientPollerStatistics;
+import org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatistics;
+
+import javax.ws.rs.client.SyncInvoker;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Utilizes the core client to poll a specified endpoint
```

Comment:
No more "core client" here...

```suggestion
 * Polls a specified HTTP(S) endpoint either synchronously or asynchronously using a JAX-RS {@link SyncInvoker}.
```